botocore/client.py

```diff
@@ -554,10 +601,12 @@ class Config(object):
     def __init__(self, region_name=None, signature_version=None,
                  user_agent=None, user_agent_extra=None,
                  connect_timeout=DEFAULT_TIMEOUT,
-                 read_timeout=DEFAULT_TIMEOUT):
+                 read_timeout=DEFAULT_TIMEOUT,
+                 s3_addressing_style=None):
```

Comment:
We should document the valid values.  It's not clear what values we support.

I'd propose at least supporting these three:

* auto (the current behavior.  Try to use virtual host when we can, path style when we can't).
* virtual_host - bucket in the hostname
* path - bucket in the URI